### DISCUSSION TITLE
Change weather of an Instance

### DISCUSSION THREAD

**Discussion by muelleel** [Minestom/Minestom#1950] [Help] [answered]
Hi,

afaik, there is currently no built in support for weather in minestom.

How can I archive getting rain with packets? I've seen the `ChangeGameStatePacket` and the `Reason.BEGIN_RAINING` but I am not sure how to send it to the Player.
The ChangeGameStatePacket class does have an constructor with an NetworkBuffer reader... how can I instantiate the packet and send it?

Thank you very much!

**Comment by GreatWyrm** [comment, marked as answer]
ChangeGameStatePacket is also a record with a constructor with parameters Reason and value. Each player has a method to send packets to them.

Here's a bit of code that will accomplish that:

```
public void sendRainToPlayer(Player player) {
    ChangeGameStatePacket packet = new ChangeGameStatePacket(Reason.BEGIN_RAINING, 0f);
    player.sendPacket(packet);
}
```

**Reply by muelleel** [reply]
Thank you, that works fine. Wonder how I've overseen that, as it works just like sending packets of other types...

Another question regarding weather: When I have some sort of shelter with blocks I expect the Rain to be blocked by these blocks. But the Rain just falls straight trough. I could not imagine this is not client-handled. It is just crazy that this seems to be server side... Or do I miss something?

I skimmed through the protocol documentation, but I did find nothing useful... Does you know how this is handled correctly?

Can I somehow get the Rain to not fall trough blocks?

**Reply by GreatWyrm** [reply]
That issue will likely be fixed by https://github.com/Minestom/Minestom/pull/1917

**Reply by muelleel** [reply]
My blocks are generated trough BlockBatches, so they are not loaded via AnvilLoader. Can I manually set the heightmap, when I place them or is this currently not implemented?

**Reply by GreatWyrm** [reply]
Currently not implemented, the heightmaps are hardcoded in `Chunk#createChunkPacket()`.

You'd have to fork Minestom and modify the source code to fix it.
